tests/test_exports.py

```diff
@@ -108,3 +108,41 @@ def test_import_base_model(tmp_path: Path):
     assert 'pydantic.fields' not in imported_modules
     assert 'pydantic.types' not in imported_modules
     assert 'annotated_types' not in imported_modules
+
+
+def test_dataclass_import(create_module):
+    @create_module
+    def module():
+        import pydantic
+
+        assert pydantic.dataclasses.__name__ == 'pydantic.dataclasses'
```

Comment:
If I revert the change in `__init__`, then `pytest tests/test_exports.py` fails here, but `pytest -k test_exports` (or just running all tests) passes because `pydantic.dataclasses` has already been imported somewhere. Doing this makes it fail more consistently without the fix:

```python
        import sys

        import pydantic

        sys.modules.pop('pydantic.dataclasses', None)
        try:
            del pydantic.dataclasses
        except AttributeError:
            pass
```

Reply:
good point, I'll change.

Reply:
fixed.